### ISSUE TITLE
Bug: CSV Upload fails and has no error message when CSV headers are incorrect

### ISSUE THREAD

**Issue by Arique1104** [StateVoicesNational/Spoke#2071]
**Describe the bug**
Although there is a message of what mandatory headers are required in the CSV Upload, we still require a clearer error message to instruct the user that the error is within their CSV headers.  Currently, all that happens is that the CSV attempts to upload, but then gives up and does not let the user know what they can do to correct the upload.

**To Reproduce**
1. Go to 'campaigns'
2. Create a CSV document where the header says cell_phone instead of cell
2. Click on 'CSV Upload'
4. See CSV Upload fail without any error messages to guide the user.

**Expected behavior**
We want to make a suggestion or have an error message pop out that provides the same information:

```
Your upload file should be in CSV format with column headings in the first row. You must include firstName, (or first_name),lastName(or last_name), and cell columns. If you include a zip column, we'll use the zip to guess the contact's timezone for enforcing texting hours. An optional column to map the contact to a CRM is external_idAny additional columns in your file will be available as custom fields to use in your texting scripts.
```

Specifically, we should add a reminder that if uploading from VAN, their header reads `cell_phone` and needs to be changed to `cell`

**Screenshots**
N/A

**Platform (please complete the following information):**
 - OS: iOS
 - Browser: Chrome
 - Desktop

**Additional context**
This bug was brought to our attention by our partner organization StateVoices and is high priority because it caused a lot of confusion during this recent election cycle.  Must have bug fixed before end of January 2022, if not sooner

**Comment by schuyler1d** [comment]
This is a different issue if it's for csv upload vs. VAN.  The CSV upload would be here:
https://github.com/MoveOnOrg/Spoke/blob/main/src/lib/parse_csv.js#L76
and
https://github.com/MoveOnOrg/Spoke/blob/main/src/extensions/contact-loaders/csv-upload/react-component.js#L93
(note the csv parsing is done client-side)

While VAN csv is done server-side (though using the same library function here)
https://github.com/MoveOnOrg/Spoke/blob/main/src/extensions/contact-loaders/ngpvan/index.js#L361
which does not look like it handles errors for the csv columns -- maybe it should (by calling failedContactLoad on an error)?

**Comment by percworld** [comment]
> https://github.com/MoveOnOrg/Spoke/blob/main/src/extensions/contact-loaders/ngpvan/index.js#L361

The ngpvan file has some interesting functions that upon first glance could solve the issue, possibly in the makeRowTransformer function.  I would like to think the handleFailedContactLoad function would not be called and a solution for the user could be reached but as I understand, we are looking to only inform them in a time-sensitive way with an error that the header must be changed in order to continue to upload a VAN csv contact.  This would need to be remedied in the ngvan/index.js file.  I figure if I'm on the wrong track now is a good time to correct that, thank you for your input.
* Also for reference, this ticket seems to be somewhat in alignment with csv uploading failing with custom headers:  [Issue 2102](https://github.com/MoveOnOrg/Spoke/issues/2102)

**Comment by schuyler1d** [comment]
if it's a just a single-(or a few) broken rows, then you can consider adding a message which is displayed on the frontend for success.  However, if certain columns are completely missing, then I assume you can't load it altogether.  The point of calling failedContactLoad (with the right arguments), is the frontend will then show the error message you say and the user can decide what to do.

**Comment by crayolakat** [comment]
This fix for this was merged into the `main` branch for the 12.2 release, but some rebasing caused the issue to not close automatically. Manually closing this issue.
